webrender/src/tiling.rs

```diff
@@ -1033,7 +1033,27 @@ impl RenderTarget for ColorRenderTarget {
 
                         // todo(gw): avoid / recycle this allocation...
                         let mut instances = Vec::new();
-                        let mut base_index = 0;
+
+                        /*
+                          GPU block layout for the text-shadow primitive.
+
+                          +-----------------------------------------+
+                          | Shadow color [1 block]                  |
+                          +-----------------------------------------+
+                          | Local space offsets [1 block per run]   |
+                          |   ...                                   |
+                          +-----------------------------------------+
+                          | Glyph offsets [1 block per TWO glyphs]  |
+                          |   ...                                   |
+                          +-----------------------------------------+
+
+                        */
+
+                        // The base glyph index is multipled by 2 here
+                        // since the glyph lookup is calculated in the
+                        // vertex shader by dividing by 2 due to the
+                        // way glyphs are packed two per GPU block.
+                        let mut base_index = 2 * prim.runs.len() as i32;
```

Comment:
this is rather hacky
I wish we had a way of defining such GPU layouts in the type system and then somehow enforcing the shaders to follow them at run-time...